runtime/ui/src/sys/linux/cuivectorfont.cpp

```diff
@@ -29,66 +31,556 @@ define_holder(ILTTexInterface, pTexInterface);
+	float f_ascent;
+	float conv, f_gap, scale;
+
+	int i_ascent, i_descent, i_gap;
+	int advance, left;
+
+	ttf = installedFontFace.GetBuffer();
+	offset = stbtt_GetFontOffsetForIndex(ttf, 0);
+	stbtt_InitFont(&info, ttf, offset);
+	scale = stbtt_ScaleForPixelHeight(&info, (float)installedFontFace.GetHeight());
+
+	if (bOk)
+	{
+		stbtt_GetFontVMetrics(&info, &i_ascent, &i_descent, &i_gap);
+		f_ascent = (float)i_ascent * scale;
+
+		// Get the sizes of each glyph.
+		bOk = GetGlyphSizes(pszChars, nLen, aGlyphMetrics,
+			sizeMaxGlyphSize, &info, scale, (int)f_ascent);
+	}
+
+	if (bOk)
+	{
+
+		// Get the size of the default character.
+		size2d_t sizeTextExtent;
+		//~ GetTextExtentPoint32(hDC, " ", 1, &sizeTextExtent);
+		//RKNSTUB
+		stbtt_GetCodepointHMetrics(&info, 32, &advance, &left);
+		conv = (float)advance * scale;
+		sizeTextExtent.cx = 32;
+		sizeTextExtent.cy = 32;
+		m_DefaultCharScreenWidth = static_cast<uint8>(conv);
+		m_DefaultCharScreenHeight = static_cast<uint8>(installedFontFace.GetHeight());
+		m_DefaultVerticalSpacing = (uint32)(((float)m_DefaultCharScreenHeight / 4.0f) + 0.5f);
+
+		// Get the average info on the characters.  The width isn't used
+		// for proportional fonts, so using an average is ok.
+		m_CharTexWidth = (uint8)25; //RKNSTUB
+		m_CharTexHeight = static_cast<uint8>(installedFontFace.GetHeight());
+
+		// Get the size our font texture should be to hold all the characters.
+		GetTextureSizeFromCharSizes(aGlyphMetrics, sizeMaxGlyphSize, nLen, sizeTexture);
+
+		// allocate the font table.  This contains the texture offsets for
+		// the characters.
+		LT_MEM_TRACK_ALLOC(m_pFontTable = new uint16[nLen * 3], LT_MEM_TYPE_UI);
+		bOk = m_bAllocatedTable = (m_pFontTable != NULL);
+	}
+
+	if (bOk)
+	{
+		// allocate the font map.  This contains the character mappings.
+		if (bMakeMap)
+		{
+			LT_MEM_TRACK_ALLOC(m_pFontMap = new uint8[256], LT_MEM_TYPE_UI);
+			bOk = m_bAllocatedMap = (m_pFontMap != NULL);
+		}
+	}
+
+	// This will be filled in with the pixel data of the font.
+	uint8* pPixelData = NULL;
+
+	// Calculate the pixeldata pitch.
+	int nPixelDataPitch = WIDTHBYTES(16, sizeTexture.cx);
+	int nPixelDataSize = nPixelDataPitch * sizeTexture.cy;
+
+	if (bOk)
+	{
+		// Allocate an array to copy the font into.
+		LT_MEM_TRACK_ALLOC(pPixelData = new uint8[nPixelDataSize], LT_MEM_TYPE_UI);
+		bOk = (pPixelData != NULL);
+		if (!bOk)
+			DEBUG_PRINT(1, ("CUIVectorFont::CreateFontTextureAndTable:  Failed to create pixeldata."));
+	}
+
+	if (bOk)
+	{
+		// set the whole font texture to pure white, with alpha of 0.  When
+		// we copy the glyph from the bitmap to the pixeldata, we just
+		// affect the alpha, which allows the font to antialias with any color.
+		uint16* pData = (uint16*)pPixelData;
+		uint16* pPixelDataEnd = (uint16*)(pPixelData + nPixelDataSize);
+		while (pData < pPixelDataEnd)
+		{
+			pData[0] = 0x0FFF;
+			pData++;
+		}
+
+		// This will hold the UV offset for the font texture.
+#ifdef _WINDOWS
```

Comment:
again linux only code... no need for if def...
and could sizeOffset not use size2d_t defined above?